price-tolerance/src/main.nr

```diff
@@ -0,0 +1,34 @@
+mod float;
+use float::{ Float, lte, div, sub, sum, abs, printlnF };
+
+global MAX_VALIDATORS = 8;
+
+fn main(
+	validators: pub [Float; MAX_VALIDATORS],
+	tolerance: pub Float,
+) {
+	// Helpers
+	let n = validators.fold(0, |acc: u32, curr: Float| {
+		if curr.0 == 0 {
+			acc
+		}
+		else {
+			acc + 1
+		}
+	});
+	let values = validators.sort_via(lte).map(|x: Float| if x.0 == 0 { (1,0) } else { x });
+
+	// Compute median
+	let median = values[(MAX_VALIDATORS - n) + n / 2];
+	printlnF(median);
+
+	// Compute reports
+	for i in 0..MAX_VALIDATORS {
+		if values[i].0 != 0 {
```

Comment:
I played around with this a little bit, in conjunction with the .map() above.

It's not clear to me how they make the "Failed constraint" error go away, but they seem to do. So good job on finding the magic combo.

Reply:
You're probably talking about the division by zero failed constraint. It's not really explicit, but in the old formula (% difference) we would execute a division by zero inside `div` function on zero values. Now that we're using % change, this is not necessary.

*Note:* since Noir always executes the bodies of if-else constructs, and uses the condition only to chose which result to discard, this means that we can't even add a check to prevent division by zero. Therefore, we change the value to 1 and still discard it (though if we're working with very small prices this might not work, now that I'm thinking about it).

Reply:
I tried to "break" the previous version with small prices, but it seemed to work. Perhaps I could have tried harder, but it doesn't matter now anyway.